docs/sources/mimir/configure/configure-zone-aware-replication.md

```diff
@@ -69,7 +69,7 @@ With a replication factor of 3, which is the default, deploy the Grafana Mimir c
 Deploying Grafana Mimir clusters to more zones than the configured replication factor does not have a negative impact.
 Deploying Grafana Mimir clusters to fewer zones than the configured replication factor can cause writes to the replica to be missed, or can cause writes to fail completely.
 
-If there are fewer than `floor(replication factor / 2)` zones with failing replicas, reads and writes can withstand zone failures.
+There can be at most `floor(replication factor / 2)` zones with failing replicas, otherwise reads and writes will fail.
```

Comment:
I think this formula is still wrong. We need a majority of zones available to accept reads and writes.

With a `replication factor = 3`, this formula gives us `floor(3 / 2) = 1` one zone that can fail -- that's correct. However, with `replication factor = 4`, this formula gives us `floor(4 / 2) = 2` two zones that can fail -- that's not correct. We need 3 zones with a replication factor of 4 (even though we don't recommend even replication factors)`. With `replication factor = 5`, this formula gives us `floor(5 / 2) = 2` two zones that can fail -- correct.

I believe the correct formula is "there can be at most `floor((replication factor - 1) / 2)`  zones with failing replicas". 

Repeating the above scenario:

With a `replication factor = 3`, this formula gives us `floor((3 - 1) / 2) = 1` one zone that can fail -- that's correct. With `replication factor = 4`, this formula gives us `floor((4 - 1) / 2) = 1` one zone that can fail -- still correct. With `replication factor = 5`, this formula gives us `floor(5 - 1) / 2) = 2` two zones that can fail -- still correct.

Please double check my reasoning and math.

Reply:
I've always thought of Mimir failure tolerance in terms of quorum because I first encountered this operating etcd. I remembered their docs being pretty good about this aspect so I dug them out:

https://etcd.io/docs/v3.3/faq/#why-an-odd-number-of-cluster-members:

> An etcd cluster needs a majority of nodes, a quorum, to agree on updates to the cluster state.
> For a cluster with n members, quorum is (n/2)+1.

https://etcd.io/docs/v3.3/faq/#what-is-failure-tolerance

Cluster Size | Majority | Failure Tolerance
-- | -- | --
1 | 1 | 0
2 | 2 | 0
3 | 2 | 1
4 | 3 | 1
5 | 3 | 2
6 | 4 | 2
7 | 4 | 3
8 | 5 | 3
9 | 5 | 4

I think the table is a really handy way of covering both sides (quorum and failure tolerance) and is easy to read for those less comfortable with the formula.